sdk/python/feast/inference.py

```diff
@@ -118,3 +125,113 @@ def update_data_sources_with_inferred_event_timestamp_col(
                     {ERROR_MSG_PREFIX} due to an absence of columns that satisfy the criteria.
                     """,
                 )
+
+
+def update_feature_views_with_inferred_features(
+    fvs: List[FeatureView], entities: List[Entity], config: RepoConfig
+) -> None:
+    """
+    Infers the set of features associated to each FeatureView and updates the FeatureView with those features.
+    Inference occurs through considering each column of the underlying data source as a feature except columns that are
+    associated with the data source's timestamp columns and the FeatureView's entity columns.
+    """
+    entity_name_to_join_key_map = {entity.name: entity.join_key for entity in entities}
+
+    for fv in fvs:
+        if not fv.features:
+            columns_to_exclude = {
+                fv.batch_source.event_timestamp_column,
+                fv.batch_source.created_timestamp_column,
+            } | {
+                entity_name_to_join_key_map[entity_name] for entity_name in fv.entities
```

Comment:
this is the line that changed from the previous version of this method. Grabbing the join_keys to exclude rather than entity name. Everything else is mainly copy and paste.

Reply:
@mavysavydav is there a reason for the move to `inference.py`? I am not against it, but it makes reviewing the PR harder. Would it make sense to at least split these two changes into different commits to make it easier on the reviewer?

Reply:
yea i agree that review becomes more difficult. I had to move this method as part of this fix b/c it needs the list of entity objects which isn't available in feature_view.py. I can split out the odfv method move into a separate PR since that's more independent if you'd like

Reply:
have taken out the odfv code move